spec/assessments/KeywordDensityAssessmentSpec.js

```diff
@@ -1,105 +1,168 @@
+		const paper = new Paper( superLongText + "keyword", { keyword: "keyword" } );
+		const researcher = new Researcher( paper );
+		const result = new KeywordDensityAssessment().getResult( paper, researcher, i18n );
+		expect( result.getScore() ).toBe( 4 );
+		expect( result.getText() ).toBe( "<a href='https://yoa.st/33v' target='_blank'>Keyphrase density</a>: The focus keyword was found 1 time. That's less than the recommended minimum of 5 times for a text of this length. <a href='https://yoa.st/33w' target='_blank'>Focus on your keyphrase</a>!" );
+	} );
+
+	it( "runs the keywordDensity on the paper with a good keyphrase density (0.5%)", function() {
+		const paper = new Paper( longText + longText + "a keyword", { keyword: "keyword" } );
+		const researcher = new Researcher( paper );
+		const result = new KeywordDensityAssessment().getResult( paper, researcher, i18n );
+		expect( result.getScore() ).toBe( 9 );
+		expect( result.getText() ).toBe( "<a href='https://yoa.st/33v' target='_blank'>Keyphrase density</a>: The focus keyword was found 1 time. This is great!" );
+	} );
+
+	it( "runs the keywordDensity on the paper with a good keyphrase density (2%)", function() {
+		const paper = new Paper( mediumText + "a keyword", { keyword: "keyword" } );
+		const researcher = new Researcher( paper );
+		const result = new KeywordDensityAssessment().getResult( paper, researcher, i18n );
+		expect( result.getScore() ).toBe( 9 );
+		expect( result.getText() ).toBe( "<a href='https://yoa.st/33v' target='_blank'>Keyphrase density</a>: The focus keyword was found 1 time. This is great!" );
+	} );
+
+	it( "runs the keywordDensity on the paper with a slightly too high keyphrase density (3.8%)", function() {
+		const paper = new Paper( longText + longText + "a keyword. Keyword. Keyword. Keyword. Keyword. Keyword. Keyword. Keyword.", { keyword: "keyword" } );
+		const researcher = new Researcher( paper );
+		const result = new KeywordDensityAssessment().getResult( paper, researcher, i18n );
 		expect( result.getScore() ).toBe( -10 );
-		expect( result.getText() ).toBe( "<a href='https://yoa.st/33v' target='_blank'>Keyphrase density</a>: 3%. This is over the advised 2.5% maximum; the keyphrase was found 2 times. <a href='https://yoa.st/33w' target='_blank'>Don't overoptimize</a>!" );
-
-		paper = new Paper( "string with the keyword  and keyword ", { keyword: "keyword" } );
-		result = new KeywordDensityAssessment().getResult( paper, factory.buildMockResearcher( {
-			getKeywordDensity: 0.5,
-			keywordCount: {
-				count: 2,
-			},
-		}, true ), i18n );
+		expect( result.getText() ).toBe( "<a href='https://yoa.st/33v' target='_blank'>Keyphrase density</a>: The focus keyword was found 8 times. That's more than the recommended maximum of 6 times for a text of this length. <a href='https://yoa.st/33w' target='_blank'>Don't overoptimize</a>!" );
+	} );
+
+	it( "runs the keywordDensity on the paper with a very high keyphrase density", function() {
```

Comment:
I think it would be good to add a comment to note that the (nonsensical) feedback text is something you wouldn't get in practice, since the assessment isn't triggered on such short texts.

Reply:
Alternatively, change the spec so that it uses a 150-word text. That way we verify that with the minimum required word count and a single keyword occurrence it provides meaningful feedback.

Reply:
Fixed.